### PR TITLE
DDCNOS-6182: handle None.get exception, now in kibana we will see rea…

### PR DESCRIPTION
**Pull request by frikit**
…l endpoint with stacktrace which cause issue

### CODE REVIEW

**Comment by frikit** [comment]
please squash and merge

**Comment by frikit** [comment]
I thinks everything should be fine with these changes

**Comment by frikit** [comment]
@danewatts please take a look

File: test/services/TransferServiceTest.scala [hunk 1]
```diff
@@ -98,6 +99,31 @@ class TransferServiceTest extends ControllerBaseSpec {
...
+  "getCurrentAndPreviousYearsEligibility" should {
+
+    "return a CurrentAndPreviousYearsEligibility" in {
+      val currentYear = 2019
+      val recipientRecord = RecipientRecord(mock[UserRecord], mock[RegistrationFormInput], List(TaxYear(currentYear)))
+      when(service.cachingService.getRecipientRecord).thenReturn(Future.successful(Some(recipientRecord)))
+      when(service.timeService.getCurrentTaxYear).thenReturn(currentYear)
+
+      val result = await(service.getCurrentAndPreviousYearsEligibility)
+      result shouldBe a[CurrentAndPreviousYearsEligibility]
```
Comment (danewatts) [inline test/services/TransferServiceTest.scala]: You want to test the values inside the object

File: app/models/CurrentAndPreviousYearsEligibility.scala [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+case class CurrentAndPreviousYearsEligibility(currentYearAvailable: Boolean, previousYears: List[TaxYear],
+                                              registrationInput: RegistrationFormInput, availableTaxYears: List[TaxYear])
+
+object CurrentAndPreviousYearsEligibility {
+
+  def apply(recipient: RecipientRecord, currentYear: Int): CurrentAndPreviousYearsEligibility = {
+    val currentYearAvailable = recipient.availableTaxYears.exists(_.year == currentYear)
+    val previousYears = recipient.availableTaxYears.filterNot(_.year == currentYear)
+
+    new CurrentAndPreviousYearsEligibility(currentYearAvailable, previousYears, recipient.data, recipient.availableTaxYears)
```
Comment (danewatts) [inline app/models/CurrentAndPreviousYearsEligibility.scala]: Dont need new keyword here

File: app/services/TransferService.scala [hunk 3]
```diff
@@ -264,18 +252,18 @@ trait TransferService {
...
   }
 
-  def saveSelectedYears(recipient: RecipientRecord, selectedYears: List[Int])(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[List[Int]] =
+  def saveSelectedYears(selectedYears: List[Int])(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[List[Int]] =
     cachingService.saveSelectedYears(selectedYears)
 
-  def updateSelectedYears(recipient: RecipientRecord, extraYear: Int, yearAvailableForSelection: Option[Int])(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[List[Int]] = {
-    updateSelectedYears(recipient, List(extraYear).filter(_ > 0), yearAvailableForSelection: Option[Int])
+  def updateSelectedYears(availableTaxYears: List[TaxYear], extraYear: Int, yearAvailableForSelection: Option[Int])(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[List[Int]] = {
+    updateSelectedYears(availableTaxYears, List(extraYear).filter(_ > 0), yearAvailableForSelection: Option[Int])
```
Comment (danewatts) [inline app/services/TransferService.scala]: Not sure why the type is on this line?

**Comment by danewatts** [comment]
Needs Squash when merging

**Pull request merged**